src/lib.rs

```diff
@@ -2470,7 +2484,10 @@ impl Bindings {
 
             // Whether we are working with C or C++ inputs.
             let is_cpp = args_are_cpp(&options.clang_args) ||
-                options.input_header.as_deref().map_or(false, file_is_cpp);
+                options
```

Comment:
Probably this should use `.any()`. That matches what `dump_preprocessed_input` is doing, and seems more sensible anyways.